kpatch-build/kpatch-build

```diff
@@ -118,7 +118,7 @@ clean_cache() {
 
 check_pipe_status() {
 	rc="${PIPESTATUS[0]}"
-	if [[ $rc = 139 ]]; then
+	if [[ "$rc" = 139 ]]; then
```

Comment:
I notice several cases like this, where the quotes aren't necessary because it's comparing a variable which isn't based on a filename.  I tried to comment on many of them below.

Reply:
As I mentioned in the commit message, I think the best way is to quote all variables even if they don't require quoting (at the moment) to prevent possible confusion by the reader and to future proof the code.

This means adding a few more quote changes in this commit but shouldn't hurt in the long run and ensures that anybody copy&pasting/adapting existing code won't introduce quoting issues. As shell's quote handling is a little fragile it's best practice to just quote everything to prevent mistakes.